docs/building.md

```diff
@@ -1,121 +1,120 @@
-    - [manual install](https://github.com/mogemimi/pomdog/wiki/How-to-Install-GYP)
-  - [ninja-build](https://github.com/ninja-build/ninja/wiki/Pre-built-Ninja-packages)
-  - tcl
-    - Using apt: `apt install tclsh`
-    - [manual install](https://www.tcl.tk/software/tcltk/)
-  - python3
-  - zlib
-    - Using apt: `apt install zlib1g-dev`
-    - Already installed on macOS by XCode.
-
-Note that we have a guide if you happen to use [WSL](howtos/wsl-build-guide.md).
-
-Ensure you have fetched the submodules first:
-
-```
-git submodule init
-git submodule update --recursive
-```
-
-The following command will ensure your environment variables are set properly and build the dependencies needed by the project:
-
-```
-./libs/verify-desktop-environment.sh
-```
-
-You can then run the Rust tests using:
-
-```
-cargo test
-```
-
-## Android development
-
-### Java 8
-
-Using any other version than 8 __will__ mess up your build and is not supported.  
-Please ensure `JAVA_HOME` points to the JDK 8 installation directory.
-
-### Setting up the Android SDK
-
-#### Through Android Studio
-
-1. Download and install Android Studio and let it install the SDK for you.
-1. In the "SDK Manager" window, find out the location of the SDK.
-1. Set `ANDROID_SDK_ROOT` and `ANDROID_HOME` (technically deprecated 🤷‍♂️) to this location and add it to your rc file.
-
-#### Manually
-
-Visit https://developer.android.com/studio/ and download the "Command line tools".  
-Install the SDK by executing:
-
-
-```
-> cd ~  
-> mkdir -p android-sdk/cmdline-tools  
-> cd android-sdk/cmdline-tools  
-> unzip {path-to.zip}  
-> # Don't forget to write these to your rc file!
-> export ANDROID_SDK_ROOT=$HOME/android-sdk
-> export ANDROID_HOME=$ANDROID_SDK_ROOT
-> $ANDROID_SDK_ROOT/cmdline-tools/tools/bin/sdkmanager --licenses
-```
-
-### Rust targets, NDK
-
-After the Android SDK is installed, run the following script that will ll install the necessary Rust targets, the Android NDK and has built-in and helpful messages to ensure you will be able to compile for Android properly:
-
-```
-./libs/verify-android-environment.sh
-````
-
-You may have to run the above script multiple times till it succeeds, once it does you can try building using:
-
+When working on application-services Rust Components it's important to set up your environment for both building the application-services library, but also for Android or iOS as appropriate to test your changes in our clients.
+
+## First time builds
+
+Building for the first time is more complicated than a typical Rust project.
+To build for an end-to-end experience that enables you to test changes in our
+client applications like Fenix and Firefox iOS, there are a number of build
+systems required for all the dependencies. The initial setup is likely to take
+a number of hours to complete.
+
+
+## Building the Rust Components
+
+*Complete this section before moving to the android/iOS build instructions.*
+
+1. Install Rust
```

Comment:
If we recommend that install via `rustup` (which is what the linked docs tell them to do) then all the version-choosing stuff should not be necessary. Simply having rustup installed should be enough, because it will read `rust-toolchain` and automagically do the right thing.